src/integrate/integrate.ts

```diff
@@ -44,15 +44,29 @@ const syncToAdmina = async (source: Source, inputs: Record<string, string>) => {
       console.log('Getting Azure AD data...')
       const azureAdData = await AzureAdSource.fetchApps(inputs)
       console.log('Registering custom service...')
-      await PromisePool
-        .for(azureAdData)
-        .withConcurrency(2) // 並列数を5に制限
-        .process(async (app: AzureAdSource.AppInfo) => {
-          await AdminaDist.registerCustomService(
-            await AzureAdTransform.transformDataToAdmina(app),
-            inputs
-          )
+      try {
+        const { results, errors } = await PromisePool
+          .for(azureAdData)
+          .withConcurrency(5) // 並列数を5に制限
+          .process(async (app: AzureAdSource.AppInfo) => {
+            await AdminaDist.registerCustomService(
+              await AzureAdTransform.transformDataToAdmina(app),
+              inputs
+            )
+          })
+        errors.forEach(error => {
+          throw
+          if (error instanceof Error) {
+            throw new Error(error.message);
+          }
         })
```

Comment:
I'd pint the all the errors and do a final throw at the end
```suggestion
        errors.forEach(error => {
          console.error('Sync failed', error);
        });
        if (errors.length > 0) {
          throw new Error("The syncing process couldn't finish completely, please check the errors");
        }
```